sklearn/utils/_metadata_requests.py

```diff
@@ -1429,23 +1429,16 @@ class attributes, as well as determining request keys from method
         # ``vars`` doesn't report the parent class attributes. We go through
         # the reverse of the MRO so that child classes have precedence over
         # their parents.
-        defaults = dict()
+        substr = "__metadata_request__"
         for base_class in reversed(inspect.getmro(cls)):
-            base_defaults = {
-                attr: value
-                for attr, value in vars(base_class).items()
-                if "__metadata_request__" in attr
-            }
-            defaults.update(base_defaults)
-        defaults = dict(sorted(defaults.items()))
```

Comment:
This looks good because the `sorting` was actually screwing the order of the MRO.

One thing that I could not figure out yet is in which data structure we add the request such that the order matter?

Reply:
Added this inline comment:

```
                    # here we "add" request values specified via those class attributes
                    # to the `MetadataRequest` object. Adding a request which already
                    # exists will override the previous one. Since we go through the
                    # MRO in reverse order, the one specified by the lowest most classes
                    # in the inheritance tree are the ones which take effect.
```

Reply:
Perfect.